packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -119,10 +127,66 @@ export class SerializedStateManager {
 		} else {
 			const { baseSnapshot, snapshotBlobs } = this.pendingLocalState;
 			this.snapshot = { baseSnapshot, snapshotBlobs };
+			this.refreshSnapshot ??= (async () => {
+				this.latestSnapshot = await getLatestSnapshotInfo(
+					this.mc,
+					this.storageAdapter,
+					supportGetSnapshotApi,
+				);
+				this.refreshPendingAttributes();
+			})();
+
 			return { baseSnapshot, version: undefined };
 		}
 	}
 
+	/**
+	 * Updates class snapshot and processedOps with the latest fetched snapshot
+	 *
+	 * @param baseSnapshot -
+	 */
+	private refreshPendingAttributes() {
+		if (this.latestSnapshot === undefined) {
+			return;
+		}
+		const snapshotSequenceNumber = this.latestSnapshot?.snapshotSequenceNumber;
+		if (this.processedOps.length === 0) {
+			// weird edge case in which we don't have to do nothing to processedOps
+			// since we don't have any.
+			return;
+		}
+		const firstProcessedOpSequenceNumber = this.processedOps[0].sequenceNumber;
+		const lastProcessedOpSequenceNumber =
+			this.processedOps[this.processedOps.length - 1].sequenceNumber;
+
+		if (snapshotSequenceNumber < firstProcessedOpSequenceNumber) {
+			// Snapshot seq number is older than our first processed op, which could mean we're fetching
+			// the same snapshot that we already have or snapshot is too old, implicating an unexpected behavior.
+			this.mc.logger.sendTelemetryEvent({
+				category:
+					snapshotSequenceNumber < firstProcessedOpSequenceNumber - 1
+						? "error"
+						: "generic",
+				eventName: "OldSnapshotFetchWhileRefreshing",
+				snapshotSequenceNumber,
+				firstProcessedOpSequenceNumber,
+			});
+			this.latestSnapshot = undefined;
+		} else if (
+			snapshotSequenceNumber >= firstProcessedOpSequenceNumber &&
```

Comment:
```suggestion
```
this condition is already covered by the ifs conditional, and it doesn't really matter, all the matters is the last op